src/core.cc

```diff
@@ -121,6 +121,12 @@
 #include "darwin.h"
 #endif
 
+#if defined (__APPLE__) && defined (__MACH__)
+#define DARWIN_DEALLOCATE_CPU_SAMPLE obj->callbacks.free = &deallocate_cpu_sample;
```

Comment:
This is a nit pick, but can you change this to a function instead of a macro? We have too many crazy C macros.

Reply:
Can you please give me an example because I am not sure how this would work?  If I understand correct, you want me to introduce a general function e.g. `deallocate_cpu` but this would mean that all OS-specific files (linux.cc, freebsd.cc, etc) should introduce this function, too!

Reply:
Yes precisely, an OS-dependent function (like `deallocate_cpu`) which can be set as the `free` callback, like this:

https://github.com/brndnmtthws/conky/blob/4d5db325d99ac5081f0e770c055562093654a6bd/src/core.cc#L579-L582

Reply:
This function can just be a no-op on other platforms btw.

This is mostly about establishing better practices/patterns. I try to make sure everything continually improves :)

Reply:
I think I can make it so that we won't have to make this change either and I will get rid of the macros!

Reply:
Ok! I am on it now!

Reply:
@brndnmtthws I got it! 😉